Build failure: trinity

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

```
nix-shell -p trinity
```

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
this derivation will be built:
  /nix/store/mm6h5r6lql79qabrprjiwrqj9pd2j8qh-trinity-1.9-unstable-2024-09-19.drv
this path will be fetched (0.19 MiB download, 1.25 MiB unpacked):
  /nix/store/0rb3lr7j4nl8822pi7xy5v6rjqqzk5f9-source
copying path '/nix/store/0rb3lr7j4nl8822pi7xy5v6rjqqzk5f9-source' from 'https://cache.nixos.org'...
building '/nix/store/mm6h5r6lql79qabrprjiwrqj9pd2j8qh-trinity-1.9-unstable-2024-09-19.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/0rb3lr7j4nl8822pi7xy5v6rjqqzk5f9-source
source root is source
Running phase: patchPhase
patching script interpreter paths in configure
configure: interpreter directive changed from "#!/bin/bash" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/bash"
patching script interpreter paths in scripts
scripts/Lindent: interpreter directive changed from "#!/bin/sh" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/sh"
scripts/analyse-dumps.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/sh"
scripts/clean-cores.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/sh"
scripts/clean-ipc.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/sh"
scripts/cppcheck.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/sh"
scripts/find.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/bash"
scripts/gen-versionh.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/bash"
scripts/hashcheck.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/sh"
scripts/killtrin.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/sh"
scripts/load-all-modules.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/bash"
scripts/paths.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/sh"
scripts/rand-vm.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/sh"
scripts/rmmod-all.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/bash"
scripts/taint.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/bash"
scripts/test-all-syscalls-parallel.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/bash"
scripts/test-all-syscalls-sequentially.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/bash"
scripts/test-multi.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/bash"
scripts/test-vm.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/bash"
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
patching script interpreter paths in ./configure
configure flags: --prefix=/nix/store/xrhrldgr4wa2a2am1ndhj380pg2birca-trinity-1.9-unstable-2024-09-19
[*] Checking system headers.
[*] Checking ipv6 headers ...  [OK] 
[*] Checking if pppox can use pppol2tpin6..  [YES] 
[*] Checking if pppox can use pppol2tv3..  [YES] 
[*] Checking if pppox can use pptp..  [YES] 
[*] Checking if llc can use LLC_OPT_PKTINFO..  [YES] 
[*] Checking if glibc headers provide termios2..  [NO] 
[*] Checking if glibc headers provide nvme ioctls..  [YES] 
[*] Checking if bpf_attr can use map_flags..  [YES] 
[*] Checking if btrfs can use btrfs_ioctl_defrag_range_args..  [YES] 
[*] Checking if glibc provides memfd_create..  [YES] 
[*] Checking if struct drm_nouveau_notifierobj_alloc is defined [YES] 
[*] Checking if struct drm_nouveau_gpuobj_free is defined [YES] 
[*] Checking header linux/caif/caif_socket.h ...  [YES] 
[*] Checking header linux/fsmap.h ...  [YES] 
[*] Checking header linux/if_alg.h ...  [YES] 
[*] Checking header linux/irda.h ...  [NO] 
[*] Checking header linux/rds.h ...  [YES] 
[*] Checking header linux/vfio.h ...  [YES] 
[*] Checking header drm/drm.h ...  [YES] 
[*] Checking header drm/i810_drm.h ...  [NO] 
[*] Checking header drm/mga_drm.h ...  [NO] 
[*] Checking header drm/r128_drm.h ...  [NO] 
[*] Checking header drm/savage_drm.h ...  [NO] 
[*] Checking header drm/exynos_drm.h ...  [YES] 
[*] Checking header sound/compress_offload.h ...  [YES] 
[*] Checking header linux/kvm.h ...  [YES] 
[*] Checking header linux/seccomp.h ...  [YES] 
[*] Checking header linux/vhost.h ...  [YES] 
[*] Checking header execinfo.h ...  [YES] 
[*] Checking header netatalk/at.h ...  [YES] 
[*] Checking header netrom/netrom.h ...  [YES] 
[*] Checking header netrose/rose.h ...  [YES] 
[*] Checking header neteconet/ec.h ...  [YES] 
[*] Checking header netax25/ax25.h ...  [YES] 
[*] Checking header netipx/ipx.h ...  [YES] 
[-] Some header definitions were missing. This is not fatal.
    It usually means you're building on an older distribution which doesn't
    have header files describing newer kernel features.
    Trinity will still compile and run, it just won't use those new features.
    Go ahead, and run 'make'
Running phase: buildPhase
build flags: -j4 SHELL=/nix/store/p79bgyzmmmddi554ckwzbqlavbkw07zh-bash-5.2p37/bin/bash
  CC    arg-decoder.o
  CC    child.o
  CC    blockdevs.o
  CC    debug.o
  CC    devices.o
  CC    ftrace.o
  CC    generate-args.o
  CC    locks.o
  CC    log-files.o
  CC    log.o
  CC    main.o
  CC    objects.o
  CC    output.o
  CC    params.o
  CC    pathnames.o
  CC    pids.o
  CC    post-mortem.o
  CC    random-syscall.o
  CC    results.o
  CC    shm.o
  CC    signals.o
  CC    stats.o
  CC    syscall.o
  CC    sysv-shm.o
  CC    tables-biarch.o
  CC    tables-uniarch.o
syscall.c: In function '__do_syscall':
syscall.c:136:1: error: bp cannot be used in 'asm' here
  136 | }
      | ^
make: *** [Makefile:115: syscall.o] Error 1
make: *** Waiting for unfinished jobs....
error: builder for '/nix/store/mm6h5r6lql79qabrprjiwrqj9pd2j8qh-trinity-1.9-unstable-2024-09-19.drv' failed with exit code 2;
       last 25 log lines:
       >   CC     locks.o
       >   CC    log-files.o
       >   CC        log.o
       >   CC      main.o
       >   CC     objects.o
       >   CC  output.o
       >   CC   params.o
       >   CC   pathnames.o
       >   CC        pids.o
       >   CC     post-mortem.o
       >   CC      random-syscall.o
       >   CC   results.o
       >   CC  shm.o
       >   CC      signals.o
       >   CC  stats.o
       >   CC    syscall.o
       >   CC  sysv-shm.o
       >   CC tables-biarch.o
       >   CC    tables-uniarch.o
       > syscall.c: In function '__do_syscall':
       > syscall.c:136:1: error: bp cannot be used in 'asm' here
       >   136 | }
       >       | ^
       > make: *** [Makefile:115: syscall.o] Error 1
       > make: *** Waiting for unfinished jobs....
       For full logs, run:
         nix log /nix/store/mm6h5r6lql79qabrprjiwrqj9pd2j8qh-trinity-1.9-unstable-2024-09-19.drv
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.37, NixOS, 25.11 (Xantusia), 25.11pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - channels(root): `"nixos, nixpkgs"`
 - nixpkgs: `/nix/store/m305283ray2vnd2n01q228lb718kykh2-nixos/nixos`

### Notify maintainers

@dezgeg 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/tr/trinity/package.nix
+++ b/pkgs/by-name/tr/trinity/package.nix
@@ -21,6 +21,9 @@ stdenv.mkDerivation (finalAttrs: {
     patchShebangs scripts
   '';
 
+  # https://salsa.debian.org/debian/trinity/-/merge_requests/2
+  env.NIX_CFLAGS_COMPILE = "-fomit-frame-pointer";
+
   enableParallelBuilding = true;
 
   installFlags = [ "DESTDIR=$(out)" ];

